spark/v3.1/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestSetCurrentSnapshotProcedure.java

```diff
@@ -219,14 +220,14 @@ public void testInvalidRollbackToSnapshotCases() {
 
     AssertHelpers.assertThrows(
         "Should reject calls without all required args",
-        AnalysisException.class,
-        "Missing required parameters",
+        IllegalArgumentException.class,
+        "Either snapshot_id or ref must be provided, not both",
```

Comment:
This error message doesn't seem right for the arguments that are being passed in. We're missing the required parameters right? Also I don't think we should change the exception that is being thrown for the existing cases (missing required parameters) because we probably want to preserve that behavior in case users rely on it. For the new failure case (ref + snapshot ID) the IllegalArgumentException is fine.

Reply:
Do you mean throw `AnalysisException` when both `snapshot_id` and `ref` are missing?

Reply:
@aokolnychyi what do you think?

Reply:
@amogh-jahagirdar, @aokolnychyi please share your thoughts here.